### DISCUSSION TITLE
How can I use the output items?

### DISCUSSION THREAD

**Discussion by goswinr** [ionide/KeepAChangelog#36] [General]
Thanks for this great tool.
I hope this is not a stupid question. 
How can I use the output items from:
https://github.com/ionide/KeepAChangelog?tab=readme-ov-file#api
Let's say I want to write the version number to some other file too.

**Comment by goswinr** [comment]
Found it:
```xml
  <Target Name="WriteChangelogVersion" AfterTargets="AfterBuild">
    <WriteLinesToFile
      File="bin/Release/ChangelogVersion.txt"
      Lines="@(CurrentReleaseChangelog)"
      Overwrite="true"/>
  </Target>
```

**Reply by baronfel** [reply]
Sorry for missing this, glad you found it! Two notes I'd suggest for your implementation though:

* Instead of hard-coding bin/Release, you can use the MSBuild Property $(OutputPath) - this is what many other Targets use for build output, and it's kept in sync with other changes you might make, like Debug/Release configuration. 
* Instead of setting Overwrite to true, which causes this write to happen every build and can break the incrementality of your build, I suggest using the [WriteOnlyWhenDifferent](https://learn.microsoft.com/en-us/visualstudio/msbuild/writelinestofile-task?view=vs-2022#task-parameters) parameter to the WriteLinesToFile Task - this will prevent additional work from being done, just kind of a best practice thing.

**Reply by goswinr** [reply]
Thanks for your help Chet @baronfel, I need this to make sure my change log version and pushed git tags are matching. My GitHub action is pushing a new release when I push a tag.  e.g.:  https://github.com/goswinr/Fesh.Rhino/blob/main/.github/workflows/release.yml#L36

I want to make sure that the tag that I pushed is matching the changelog. It still feels a bit clumsy, let me know if there is a more elegant way to do this,
